front/app/utils/permissions/rules/projectPermissions.ts

```diff
@@ -18,16 +20,22 @@ definePermissionRule('project', 'reorder', (_project: IProjectData, user) => {
 });
 
 export function canModerateProject(
-  projectId: string | null | undefined,
+  project: IProjectData,
   user: IUser | undefined
 ) {
-  if (projectId) {
-    return isAdmin(user) || isProjectModerator(user, projectId);
-  }
+  const projectId = project.id;
+  const projectFolderId = project.attributes.folder_id;
 
-  return isAdmin(user);
+  if (!user) return false;
+
+  return (
+    isAdmin(user) ||
+    (typeof projectFolderId === 'string' &&
+      userModeratesFolder(user, projectFolderId)) ||
+    isProjectModerator(user, projectId)
+  );
```

Comment:
Extending canModerateProject to include "can moderate folder that contains project"

Reply:
I am not sure we should do this given the system we have in place. Let's discuss.

Reply:
The problem is that when you assign someone to be a folder moderator, they automatically become the moderator of all the projects in that folder, as well. But technically, you can remove some of these projects by un-assigning the user from the project. Then this project will no longer appear under "your projects" for example. Perhaps we should make sure it is not possible to unassign folder moderators from the project 🤔

Reply:
Hm.. a bit of an edge case but indeed, not great UX. I do think it should not be possible to remove folder moderators from projects in folder that they are moderating because this makes very little sense. If we don't allow this in the first place, we will also solve the issue above. In any case, if it's the case now, I don't think fixing it is within the scope of this PR.